bmc/firmware.go

```diff
@@ -5,25 +5,27 @@ import (
 	"fmt"
 	"io"
 
+	"github.com/bmc-toolbox/bmclib/v2/constants"
+	bconsts "github.com/bmc-toolbox/bmclib/v2/constants"
 	bmclibErrs "github.com/bmc-toolbox/bmclib/v2/errors"
 
 	"github.com/hashicorp/go-multierror"
 	"github.com/pkg/errors"
 )
 
-// FirmwareInstaller defines an interface to install firmware updates
+// FirmwareInstaller defines an interface to upload and initiate a firmware install
 type FirmwareInstaller interface {
 	// FirmwareInstall uploads firmware update payload to the BMC returning the task ID
 	//
 	// parameters:
 	// component - the component slug for the component update being installed.
-	// applyAt - one of "Immediate", "OnReset".
+	// operationsApplyTime - one of the OperationApplyTime constants
```

Comment:
Why not make that a type instead of constants of `string`?

Reply:
Yep, that is the intention going ahead in the newer interface, for now we don't want to break the current interface.